src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,42 @@
 package org.example.bookstore.service.impl;
 
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import org.example.bookstore.dto.BookDto;
+import org.example.bookstore.dto.CreateBookRequestDto;
+import org.example.bookstore.mapper.BookMapper;
 import org.example.bookstore.model.Book;
 import org.example.bookstore.repository.BookRepository;
 import org.example.bookstore.service.BookService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+@AllArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        Optional<List<Book>> booksOptional = Optional.ofNullable(bookRepository.findAll());
+        return booksOptional
```

Comment:
```suggestion
       return  bookRepository.findAll();
```
It will be empty list if there is no books

Reply:
> It will be empty list if there is no books

You have to return BookDTO